app/prisma/schema.prisma

```diff
@@ -613,22 +615,76 @@ model UsageLog {
 
     inputTokens  Int
     outputTokens Int
+    inputCost    Decimal @default(0)
+    outputCost   Decimal @default(0)
     cost         Float
     type         UsageType @default(EXTERNAL)
     billable     Boolean   @default(true)
+    baseModel  String?
 
     fineTuneId String?   @db.Uuid
     fineTune   FineTune? @relation(fields: [fineTuneId], references: [id], onDelete: SetNull)
-
-    projectId String?  @db.Uuid
-    project   Project? @relation(fields: [projectId], references: [id], onDelete: Cascade)
+    projectId  String?  @db.Uuid
+    project    Project? @relation(fields: [projectId], references: [id], onDelete: Cascade)
+    invoiceId String?  @db.Uuid
+    invoice  Invoice? @relation(fields: [invoiceId], references: [id], onDelete: SetNull)
 
     createdAt DateTime @default(now())
 
     @@index([fineTuneId, createdAt, type])
     @@index([projectId, createdAt, type])
 }
 
+enum InvoiceStatus {
+    PENDING
+    PAID
+    CANCELLED
+    REFUNDED
+}
+model Invoice {
+    id String @id @default(uuid()) @db.Uuid
+    amount         Decimal @default(0)
+    status         InvoiceStatus @default(PENDING)
+    slug           String
+    billingPeriod  String? // MM YYYY
+    paidAt         DateTime?
+
+    description Json? // Array of objects: { [key: string]: string }
+    paymentId String?
+
+    usageLogs UsageLog[]
+    creditAdjustments CreditAdjustment[]
+
+    projectId String  @db.Uuid
+    project   Project @relation(fields: [projectId], references: [id], onDelete: Cascade)
+
+    createdAt DateTime @default(now())
+
+    @@index([projectId, createdAt])
+}
+
+enum CreditAdjustmentType {
```

Comment:
In the latest commit, I added this CreditAdjustmentType enum to the schema. What do you think @corbt , do we need it?

Reply:
We could remove `type` and use only `invoiceId` on `CreditAdjustment` table (we already have this relation) to associate negative credit adjustments with an invoice.
CreditAdjustmentType may be useful if we have different types of credit adjustments, and if we want to explicitly track them.